src/GitHub/App/Token/Generate.hs

```diff
@@ -88,36 +109,97 @@ generateInstallationTokenScoped
   -> InstallationId
   -> m AccessToken
 generateInstallationTokenScoped create creds installationId = do
-  jwt <- signJWT expiration issuer creds.privateKey
-
   req <-
-    liftIO
-      $ parseRequest
-      $ "POST https://api.github.com/app/installations/"
+    githubRequest
+      $ "/app/installations/"
       <> show installationId.unwrap
       <> "/access_tokens"
 
   -- Avoid encoding to "{}", which causes a 500
   let setBody = if create == mempty then id else setRequestBodyJSON create
 
+  appHttpJSON AccessTokenHttpError AccessTokenJsonDecodeError creds
+    $ setRequestMethod "POST"
+    $ setBody req
+
+generateOwnerTokenScoped
+  :: MonadIO m
+  => CreateAccessToken
+  -> AppCredentials
+  -> Owner
+  -> m AccessToken
+generateOwnerTokenScoped create creds owner = do
+  installation <- getInstallation creds $ case (create.repositories, owner) of
+    (repo : _, Org org) -> "/repos/" <> org <> "/" <> repo
+    (repo : _, User username) -> "/repos/" <> username <> "/" <> repo
```

Comment:
Is it possible that these `: _`s could be throwing away repos that consumers might expect to have been included?

If not, can the types be re-worked to make those states impossible (or at least not appear in the `case` statement)?

If so, is that okay? Could it cause headache?

Reply:
It's a little confusing, but I'm mimicking the behavior of `actions/create-github-app-token`, which is:

If `repositories` is present as a scope, use `repositories.first` to get the installation id -- but we are still including all of them when it is used as a scope.

So the `_` are discarding other repositories here (and we're ignoring `repository_ids` entirely), but that all still gets used when getting the actual access-token.

Reply:
This logic has a comment in the action, albeit one that that describes the _what_ more than the _why_. I think I'll add a (better) comment here.

Reply:
Makes sense, thanks Pat.